### ISSUE TITLE
DateTimePicker picker control for time

### ISSUE THREAD

**Issue by DJViking** [edvin/tornadofx-controls#8]
The DateTimePicker does not have a TimePicker, just a DatePicker. The button opens the same calendar as the DatePicker.

**Comment by edvin** [comment]
That's right, it currently only allows you to pick the date and you have to write the time yourself. I did it like this because it's more convenient IMO. Picking the time just takes more time (pun intented), but for date you might need to consult a calendar. There is only one button, so this shouldn't be confusing, but I'd be up for adding a time picker in form of two spinners perhaps, if that's what you're envisioning?

Any feedback or sketch you can provide with regards to layout suggestions are most welcome :)

**Comment by DJViking** [comment]
The project ControlsFX started with a TimePicker, but it did not make it.
https://bitbucket.org/controlsfx/controlsfx/pull-requests/522/new-control-timepicker/diff

There is even a JDK feature request for a TimePicker.
https://bugs.openjdk.java.net/browse/JDK-8090397
Java has LocalTime, LocalDate, LocalDateTime. It should definitely have JavaFX control for each of these.

The project JFoenix has a similar TimePicker implementation:
https://github.com/jfoenixadmin/JFoenix
![timepicker](https://user-images.githubusercontent.com/3070076/31005107-949feca4-a4f7-11e7-85d2-b16e1f471046.gif)

Such a similar TimePicker could be implemented as a second popup after choosing the date.

**Comment by edvin** [comment]
Certainly possible, but I think we're better off reimplementing the date picker to support grater freedom and control over the functionality and design. The DateTimePicker in it's current form is simply a hack around the stock DatePicker. I don't have time to look at this right now, but a PR would be most welcome :)

**Comment by iameddz** [comment]
I would also look forward to a TimePicker!

**Comment by DJViking** [comment]
The time picker could be implemented as sliders, perhaps something similar like the Android date time picker.
![Screenshot_20190314-203309](https://user-images.githubusercontent.com/3070076/54554619-2d558b80-49b5-11e9-8d72-fcbbcd9d19f8.png)

**Comment by edvin** [comment]
That design works on mobile but seems quite cumbersome to use on a desktop IMO.

**Comment by owan1024** [comment]
Works fine with one small comment. Edit of the time gets lost if you re-pick the date. Think this would be solved for example by using the editor text string (or perhaps adding a commit ).

replace: time = localDateTime.toLocalTime();
with:      time = LocalDateTime.parse(this.getEditor().getText(), formatter).toLocalTime();
